multy_test/test_golos_transaction.cpp

```diff
@@ -0,0 +1,265 @@
+/* Copyright 2018 by Multy.io
+ * Licensed under Multy.io license.
+ *
+ * See LICENSE for details
+ */
+
+#include "multy_core/src/golos/golos_account.h"
+#include "multy_core/binary_data.h"
+
+#include "multy_core/account.h"
+#include "multy_core/big_int.h"
+#include "multy_core/properties.h"
+#include "multy_core/transaction.h"
+
+#include "multy_core/src/api/big_int_impl.h"
+#include "multy_core/src/api/account_impl.h"
+#include "multy_core/src/api/key_impl.h"
+#include "multy_core/src/u_ptr.h"
+#include "multy_core/src/utility.h"
+
+#include "multy_test/utility.h"
+#include "multy_test/value_printers.h"
+
+#include "multy_core/golos.h"
+
+#include "gtest/gtest.h"
+
+namespace
+{
+using namespace multy_core::internal;
+using namespace test_utility;
+} // namespace
+
+GTEST_TEST(GolosTransactionTest, SmokeTest_public_api)
+{
+    AccountPtr account;
+
+    HANDLE_ERROR(make_account(
+                     BLOCKCHAIN_GOLOS,
+                     "5JpDgood17pE47zB6pDJixg9Sw47QiHcQ9qCc3MeKYoYzRiMcnF",
+                     reset_sp(account)));
+
+    TransactionPtr transaction;
+    HANDLE_ERROR(make_transaction(account.get(), reset_sp(transaction)));
+    ASSERT_NE(nullptr, transaction);
+
+    {
+        Properties* properties = nullptr;
+        const BigInt ref_block_num(14918824);
```

Comment:
According to Golos sources (https://github.com/GolosChain/golos/blob/a55653892d9325760edc46ad077eca540a561fe7/libraries/protocol/include/golos/protocol/transaction.hpp#L12):
```
        struct transaction {
            uint16_t ref_block_num = 0;
            uint32_t ref_block_prefix = 0;
```
'ref_block_num' is 16 bit, `ref_block_prefix` is 32 bit, hence both do fit into `int32_t` and there is no need in `BigInt`.
However, given nature of `ref_block_prefix` (part of a block hash), it makes sense to provide that as `BinaryData`. It is up to the TX implementation to serialize it in a number format.

Reply:
Above is incorrect, please use int32 for block_num and BinaryData for block_hash